### PR TITLE
feature: Initial cosmos keplr provider

### LINKED ISSUE #12: COSMOS wallets: Add Keplr support

### PR DESCRIPTION
**Pull request by gbarkhatov**
Closes #12 

We need to discuss the `RPC` link and `chainId` as params: https://github.com/babylonlabs-io/bbn-wallet-connect/issues/56

We might save and get the chainId from the wallet interface (we can get it from Keplr when we're connecting)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -88,5 +90,6 @@
     "extends": [
       "plugin:storybook/recommended"
     ]
-  }
+  },
+  "packageManager": "pnpm@9.13.0+sha512.beb9e2a803db336c10c9af682b58ad7181ca0fbd0d4119f2b33d5f2582e96d6c0d93c85b23869295b765170fbdaa92890c0da6ada457415039769edf3c959efe"
```
[the remark this answers is not in the document]
Comment (gbarkhatov) [reply package.json:94]: Automatically added on my side, can remove it

File: src/core/wallets/bbn/keplr/index.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import { KeplrProvider } from "./provider";
+
+const metadata: WalletMetadata<BBNProvider> = {
+  id: "keplr",
+  name: "Keplr",
+  icon: logo,
+  docs: "https://www.keplr.app/",
+  wallet: "keplr",
+  createProvider: (wallet) => new KeplrProvider(wallet),
+  networks: [Network.MAINNET, Network.SIGNET],
```
[the remark this answers is not in the document]
Comment (gbarkhatov) [reply src/core/wallets/bbn/keplr/index.ts:14]: Signet at the moment is an application mode on `simple-staking`.
If you think `X` wallet is working on Mainnet and Signet and `Y` does not work on Signet, you can disable the needed wallet.

Same way, we can control the `on/off` on Cosmos wallets. Whether or not wallet X should work when BTC Mainnet mode is on, or BTC Signet mode is on.

Maybe Keplr is tested and works good on BTC Mainnet, but BananaCosmosWallet are in testing phase and should be enabled on Signet only

File: src/core/wallets/bbn/BBNProvider.ts
[the remark this answers is not in the document]
Comment (gbarkhatov) [reply src/core/wallets/bbn/BBNProvider.ts:1]: @jrwbabylonlab https://github.com/babylonlabs-io/bbn-wallet-connect/issues/55

File: src/core/wallets/bbn/keplr/provider.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
+import { Window as KeplrWindow } from "@keplr-wallet/types";
+import { Buffer } from "buffer";
+
+import { WalletInfo } from "@/core/types";
+import { BBNProvider } from "@/core/wallets/bbn/BBNProvider";
+
+import { OfflineAminoSigner, OfflineDirectSigner } from "@keplr-wallet/types/src/cosmjs";
+import { SigningStargateClient, SigningStargateClientOptions } from "@cosmjs/stargate";
+
+const DEFAULT_RPC = "https://cosmoshub.validator.network:443";
```
[the remark this answers is not in the document]
Comment (gbarkhatov) [reply src/core/wallets/bbn/keplr/provider.ts:10]: We have #54 related to bbn config and rpc/chainid

[the remark this answers is not in the document]
Comment (gbarkhatov) [reply src/core/wallets/bbn/keplr/provider.ts]: Marking as resolved https://github.com/babylonlabs-io/bbn-wallet-connect/pull/51/commits/6aac2f6ad49b624b14637203a88f0ec03c1fccdc

**Review by totraev — APPROVED** [review]

File: src/core/chains/bbnDevnet.ts
[the remark this answers is not in the document]
Comment (gbarkhatov) [reply src/core/chains/bbnDevnet.ts:1]: We might use these configs (`bbn-devnet`, `bbn-testnet`, `bbn-x`) later on when tests are added to this repo

**Pull request merged**